openfl/federated/data/sources/pytorch/s3_folder.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2020-2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+from urllib.parse import urlparse
+
+import boto3
+
+from openfl.federated.data.sources.pytorch.folder_dataset import FolderDataset, LabelMapper
+
+
+class S3Folder(FolderDataset):
+    def __init__(
+        self,
+        uri,
+        label_mapper: LabelMapper,
+        endpoint=None,
+        access_key_env_name=None,
+        secret_key_env_name=None,
+        transform=None,
+    ):
+        """
+        Args:
+            uri (str): URI to the S3 object.
+            label_mapper (LabelMapper): LabelMapper object to map class names to indices.
+            transform (callable, optional): Transformations to apply to images.
```

Comment:
could you update the doc string to match the actual parameters?

Reply:
done